Vagrantfile

```diff
@@ -0,0 +1,170 @@
+require 'json'
+require 'open-uri'
+require 'vagrant-aws'
+
+home_dir="/home/ubuntu"
+current_branch=`git rev-parse --abbrev-ref HEAD`
+latest_head_commit=`git rev-parse HEAD`
+latest_origin_commit=`git rev-parse origin/#{current_branch}`
+clean_tree=system('git status | grep "nothing to commit"')
+ssh_identities=`ssh-add -l`
+stop_hours = 48  # if STOP_HOURS ENV is not specified, stop the instance after 2 days of running
+if !ENV['STOP_HOURS'].nil?
+  stop_hours = ENV['STOP_HOURS']
+end
+
+# Only run these checks on `vagrant up`
+if ARGV[0] == "up"
+  if ssh_identities.strip == 'The agent has no identities.'
+    raise "No ssh identities are loaded, run `ssh-add` to load the private key that is allowed to clone the a2 repo!"
+  end
+  if !clean_tree
+    puts %q(
+      !!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!
+      ! You have uncommitted changes that won't exist when we do the git clone on the remote EC2 instance !
+      !!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!
+    )
+  end
+  if latest_head_commit != latest_origin_commit
+    puts %q(
+      !!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!
+      ! You have unpushed commits that won't exist when we do the git clone on the remote EC2 instance !
+      !!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!
+    )
+  end
+
+  if ENV['AWS_ACCESS_KEY_ID'].nil? || ENV['AWS_SECRET_ACCESS_KEY'].nil?
+    raise "ENV variables AWS_ACCESS_KEY_ID and AWS_SECRET_ACCESS_KEY must be defined for this, aborting..."
+  end
+
+  if ENV['GITHUB_TOKEN'].nil?
+    raise "ENV variable GITHUB_TOKEN must be defined for this, aborting..."
+  end
+
+  if ENV['AWS_SSH_KEY_NAME'].nil?
+    raise "ENV variable AWS_SSH_KEY_NAME must be defined for this. See README.md for more details. Aborting..."
+  end
+end
+
+def hab_version_from_manifest
```

Comment:
this is very Automate specific. We shouldn't check this in since this is an open source project. 😄

Reply:
maybe it's best to just get latest hab ver rather than specific ver?

Reply:
That or use what ci-studio-commons gives you. either way works!

Reply:
We use the latest hab now.  There is no longer a dependency on automate hab version.

Reply:
I would like to delete this still, but we can do it latter.